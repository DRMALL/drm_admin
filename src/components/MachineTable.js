

import Table from 'antd/lib/table'
import React from 'react'
import TableTime from './TableTime'
import TableTextHidden from './TableTextHidden'
// import TableAddress from './TableAddress'
import { browserHistory } from 'react-router'
import Modal from 'antd/lib/modal'
import machineDelById from '../actions/machineDelById'
import dispatch from '../actions/dispatch'

export default props => {
  const { machineArr, machineModal } = props.state.machine
  const columns =[{
    title: '设备名称',
    dataIndex: 'name',
    key: 'name',
    width: '10%'
  },{
    title: '设备编号',
    dataIndex: 'number',
    key: 'number',
    width: '10%'
  },{
    title: '排量',
    dataIndex: 'cc',
    key: 'cc',
    width: '10%'
  },{
    title: '压力',
    dataIndex: 'pressure',
    key: 'pressure',
    width: '10%'
  },{
    title: '燃料',
    dataIndex: 'combustible',
    key: 'combustible',
    width: '10%',
  },{
    title: '描述',
    dataIndex: 'description',
    key: 'description',
    width: '10%',
    render: text => TableTextHidden( text, 10)
  },{
    title: '负责人',
    dataIndex: 'incharges',
    width: '10%',
    render: text => <div>{text.slice(0,4).join(', ')}</div>
  },{
    title: '所在地',
    dataIndex: 'address',
    key: 'address',
    width: '10%',
    // render: text => TableAddress(text)
  },{
    title: '添加时间',
    dataIndex: 'createdAt',
    key: 'createdAt',
    width: '10%',
    sorter: (a,b) => new Date(a.createdAt) - new Date(b.createdAt),
    render: text => TableTime(text, 15)
  },{
    title: '操作',
    key: 'action',
    width: '10%',
    render: (text, record) =>(
      <div className='machine-table-action' >
        <div className='machine-table-left' data-machine-id={ text._id }
             onClick={ navToMachineEdit } >
          <i className="iconfont icon-compile" style={{ fontSize:20 }} ></i>
        </div>
        <div className='machine-table-right' data-machine-id={ text._id }
             onClick={ navToMachineControl } >
          <i className="iconfont icon-details" ></i>
        </div>
      </div>
      )
  },]
  return(
    <div>
      <Table dataSource={ machineArr } columns={ columns } rowKey='_id' />
      <Modal title='提示' visible={ machineModal } okText='确定' cancelText='取消'
             onOk={ machineDelById } onCancel={ cancleDelAuth }  >
        <p>确定删除此设备？</p>
      </Modal>
    </div>
    )
}

// function delMachinexx(e){
//   dispatch('MACHINE_GET_DEL_MACHINE_ID', e.currentTarget.dataset.machineId)
// }

function cancleDelAuth(){
  dispatch('MACHINE_CANCLE_DEL_MACHINE')
}

function navToMachineEdit(e){
  const { machineId } = e.currentTarget.dataset
  localStorage.setItem('machineId', machineId)
  browserHistory.push('/machine/edit')
}

function navToMachineControl(e){
  const { machineId } = e.currentTarget.dataset
  localStorage.setItem('machineId', machineId)
  browserHistory.push('/machine/control')
}

//设备删除
        // <div className='message-arr-item' onClick={ delMachinexx } data-machine-id={ record._id }  >
        //   <i className="iconfont icon-shanchu message-del-icon"  ></i>
        // </div>
