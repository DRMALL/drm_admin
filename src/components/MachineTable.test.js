

import MachineTable from './MachineTable'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

it('MachineTable test', () => {
  const state = { machine: {
    machineArr: [{name: '张三', number: 'a1254', cc: '354', pressure: '30mpa', combustible: '汽油', _id: '4541342dsf', incharges:['daes', 'sdse']  },
    {name: '张三', number: 'a1254', cc: '354', pressure: '30mpa', combustible: '汽油',_id: '4541342dssdr', incharges:['daes', 'sdse']   }] }, machineModal: true }
  const wrapper = render(<MachineTable state={ state } />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
