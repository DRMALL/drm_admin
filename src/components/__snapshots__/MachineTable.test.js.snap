// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MachineTable test 1`] = `
<div>
  <div
    class="ant-table-wrapper"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <div
          class="ant-table ant-table-large ant-table-scroll-position-left"
        >
          <div
            class="ant-table-content"
          >
            <div
              class="ant-table-body"
            >
              <table
                class=""
              >
                <colgroup>
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                  <col
                    style="width:10%;min-width:10%;"
                  />
                </colgroup>
                <thead
                  class="ant-table-thead"
                >
                  <tr>
                    <th
                      class=""
                    >
                      <span>
                        设备名称
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        设备编号
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        排量
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        压力
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        燃料
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        描述
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        负责人
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        所在地
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        添加时间
                        <div
                          class="ant-table-column-sorter"
                        >
                          <span
                            class="ant-table-column-sorter-up off"
                            title="↑"
                          >
                            <i
                              class="anticon anticon-caret-up"
                            />
                          </span>
                          <span
                            class="ant-table-column-sorter-down off"
                            title="↓"
                          >
                            <i
                              class="anticon anticon-caret-down"
                            />
                          </span>
                        </div>
                      </span>
                    </th>
                    <th
                      class=""
                    >
                      <span>
                        操作
                      </span>
                    </th>
                  </tr>
                </thead>
                <tbody
                  class="ant-table-tbody"
                >
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      张三
                    </td>
                    <td
                      class=""
                    >
                      a1254
                    </td>
                    <td
                      class=""
                    >
                      354
                    </td>
                    <td
                      class=""
                    >
                      30mpa
                    </td>
                    <td
                      class=""
                    >
                      汽油
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div>
                        daes, sdse
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="machine-table-action"
                      >
                        <div
                          class="machine-table-left"
                          data-machine-id="4541342dsf"
                        >
                          <i
                            class="iconfont icon-compile"
                            style="font-size:20px;"
                          />
                        </div>
                        <div
                          class="machine-table-right"
                          data-machine-id="4541342dsf"
                        >
                          <i
                            class="iconfont icon-details"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                  <tr
                    class="ant-table-row  ant-table-row-level-0"
                  >
                    <td
                      class=""
                    >
                      <span
                        class="ant-table-row-indent indent-level-0"
                        style="padding-left:0px;"
                      />
                      张三
                    </td>
                    <td
                      class=""
                    >
                      a1254
                    </td>
                    <td
                      class=""
                    >
                      354
                    </td>
                    <td
                      class=""
                    >
                      30mpa
                    </td>
                    <td
                      class=""
                    >
                      汽油
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div>
                        daes, sdse
                      </div>
                    </td>
                    <td
                      class=""
                    />
                    <td
                      class=""
                    />
                    <td
                      class=""
                    >
                      <div
                        class="machine-table-action"
                      >
                        <div
                          class="machine-table-left"
                          data-machine-id="4541342dssdr"
                        >
                          <i
                            class="iconfont icon-compile"
                            style="font-size:20px;"
                          />
                        </div>
                        <div
                          class="machine-table-right"
                          data-machine-id="4541342dssdr"
                        >
                          <i
                            class="iconfont icon-details"
                          />
                        </div>
                      </div>
                    </td>
                  </tr>
                </tbody>
              </table>
            </div>
          </div>
        </div>
        <ul
          class="ant-pagination ant-table-pagination"
          unselectable="unselectable"
        >
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-prev"
            tabindex="0"
            title="上一页"
          >
            <a />
          </li>
          <li
            class="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
            tabindex="0"
            title="1"
          >
            <a>
              1
            </a>
          </li>
          <li
            aria-disabled="true"
            class="ant-pagination-disabled ant-pagination-next"
            tabindex="0"
            title="下一页"
          >
            <a />
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
